[IT-2886] Address dependabot alert

ConsoleCatzirl opened the pull request:
I created commit https://github.com/Sage-Bionetworks-IT/lambda-send-budget-alert/commit/85672cc11bfb7a997c81c33a392f9bd68c1e1d3c by running pipenv update -d but then the tests failed with the traceback I posted in jira. The error seems to be coming from an upgrade of urllib3 breaking botocore. I resolved the error in commit https://github.com/Sage-Bionetworks-IT/lambda-send-budget-alert/commit/70f09cf8a65b5a9147abfb65d0b94bafc1f23491 by pinning urllib3 to a version compatible with botocore, but then replaced that commit in this PR with commit https://github.com/Sage-Bionetworks-IT/lambda-send-budget-alert/commit/dfec561da4e79ade5d6bcb887e81e57e9cbac31d which no longer attempts to upgrade urllib3 to a version incompatible with botocore (and it also made the command `pipenv update -d` run much quicker).

ConsoleCatzirl committed:
[IT-2886] dependabot alert

Update dependencies to address a dependabot alert

zaro0508 reviewed (commented):
i don't understand the problem and what this PR is trying to resolve.  The linked jira issue did not help clarify it for me.

ConsoleCatzirl commented:
This PR is trying to resolve these dependabot alerts for requests and flask: https://github.com/Sage-Bionetworks-IT/lambda-send-budget-alert/security/dependabot?q=is%3Aclosed+resolution%3Afix-started

ConsoleCatzirl committed:
[IT-2886] Loosen dependency constraints

Allow pipenv to update minor versions of listed dependencies (e.g.
synapseclient 2.6 -> 2.7)

Newer versions of packages have more consistent deep dependencies,
avoiding the need to add an additional constraint on urllib3, and
speeding up the time pipenv takes to resolve dependencies from 15
minutes to 90 seconds.

ConsoleCatzirl commented:
Loosen constraints instead of adding a new constraint on urllib3

zaro0508 commented:
i don't understand how this fixes a "urllib3 dependency mismatch".  in the issue you say `..  and also resolved the need to pin urllib3.`  I don't see any pinning or pin removal of urllib3.  Could you please explain?

ConsoleCatzirl commented:
Yeah, I created commit 85672cc by running `pipenv update -d` but then the tests failed with the traceback I posted in jira. The error seems to be coming from an upgrade of urllib3 breaking botocore. I resolved the error in commit 70f09cf by pinning urllib3 to a version compatible with botocore, but then replaced that commit in this PR with commit dfec561 which no longer attempts to upgrade urllib3 to a version incompatible with botocore (and also running `pipenv update -d` was much quicker).

zaro0508 reviewed (approved):
your last comment in this PR should be the commit message for this change.
